### PR TITLE
Send userId field if available when reporting notification events

### PR DESCRIPTION
**Pull request by lukabratos**
We want to send `userId` to make it easy to support the webhooks we are planning to release.

### CODE REVIEW

File: Sources/ReportEventType.swift [hunk 1]
```diff
@@ -20,15 +22,17 @@ struct DeliveryEventType: ReportEventType {
     let event: String
     let publishId: String
     let deviceId: String
+    let userId: String
```
Comment (lukabratos) [inline Sources/ReportEventType.swift]: Right now I'm setting `userId` to an empty string if the payload doesn't contain `userId`.
I can make this optional as well. What do you think @luismfonseca ?

File: Sources/EventTypeHandler.swift [hunk 2]
```diff
@@ -44,13 +46,15 @@ struct EventTypeHandler {
             return nil
         }
 
+        let userId = UserId(userInfo: userInfo).id ?? ""
```
Comment (lukabratos) [inline Sources/EventTypeHandler.swift]: See the comment below.

File: Sources/ReportEventType.swift (hunk 1 above)
Comment (luismfonseca) [reply Sources/ReportEventType.swift]: I think it makes more sense to have it as optional.

File: Sources/EventTypeHandler.swift (hunk 2 above)
Comment (luismfonseca) [reply Sources/EventTypeHandler.swift]: Basically, the `?? ""` isn't needed if we just accept that userId is optional

**Pull request merged**